tests/project/test_brownie_config.py

```diff
@@ -128,3 +131,42 @@ def test_validate_cmd_settings():
 def test_raise_validate_cmd_settings(invalid_setting):
     with pytest.raises(TypeError):
         _validate_cmd_settings(invalid_setting)
+
+
+@pytest.fixture
+def settings_proj_with_env(testproject):
+    """Creates a config file with a dotenv in the testproject root folder and loads it manually."""
+
+    test_dotenv_config = """
+    FOO=bar
+    THE_ANSWER=42
+    """
+
+    env_filename = "".join(random.choice(string.ascii_lowercase) for i in range(10))
+    test_brownie_config = f"""
+    dotenv: {env_filename}
+    networks:
+        default: development
+        development:
+            gas_limit: 6543210
+    """
+    env_path = testproject._path.joinpath(env_filename)
+    with env_path.open("w") as fp:
+        fp.write(test_dotenv_config)
```

Comment:
Should probably use `TemporaryFile`, but I'd like to have a file that's a sibling to the config, since I think that's the most common way the `dotenv` directive will be used.